router/router.go

```diff
@@ -141,8 +152,11 @@ func (r *routeSet) ProcessRoute(uri *url.URL) (newurl string, requirePermission
 }
 
 func (r *routeSet) GetProxy(req *ms_http.Request) (*url.URL, error) {
-	route := r.match(req.URL)
-	if route == nil || !route.target.useProxy {
+	if r.targetHash[req.URL.Host] == nil {
+		fmt.Errorf("Nil target pointer found in hash for host %v", req.URL.Host)
```

Comment:
Messages to stdout (which is where fmt emits) will get discarded in production, so it's very unlikely that somebody would see this message when it mattered. We need to use the real logging system. It's awkward that routeSet doesn't have access to the higher level data structure "Server". What I would do here is to change GetProxy so that it reads as follows:
```go
 func (r *routeSet) GetProxy(errLog *log.Logger, req *ms_http.Request) (*url.URL, error) {
    // ...
    errLog.Errorf("Nil target pointer found in hash for host %v", req.URL.Host)
    // ...
}
```
You'll also need to change the Router interface to match.